SPECS/libvirt/libvirt.spec

```diff
@@ -1,654 +1,1970 @@
-Requires:       %{name}-libs = %{version}-%{release}
+    %if %{with_storage_rbd}
+%package daemon-driver-storage-rbd
+Summary: Storage driver plugin for rbd
+Requires: libvirt-daemon-driver-storage-core = %{version}-%{release}
+Requires: libvirt-libs = %{version}-%{release}
 
 %description daemon-driver-storage-rbd
 The storage driver backend adding implementation of the storage APIs for rbd
 volumes using the ceph protocol.
+    %endif
+
+    %if %{with_storage_zfs}
+%package daemon-driver-storage-zfs
+Summary: Storage driver plugin for ZFS
+Requires: libvirt-daemon-driver-storage-core = %{version}-%{release}
+Requires: libvirt-libs = %{version}-%{release}
+# Support any conforming implementation of zfs
+Requires: /sbin/zfs
+Requires: /sbin/zpool
+
+%description daemon-driver-storage-zfs
+The storage driver backend adding implementation of the storage APIs for
+ZFS volumes.
+    %endif
 
-%package daemon-driver-storage-scsi
-Summary:        Storage driver plugin for local scsi devices
+%package daemon-driver-storage
+Summary: Storage driver plugin including all backends for the libvirtd daemon
+Requires: libvirt-daemon-driver-storage-core = %{version}-%{release}
+Requires: libvirt-daemon-driver-storage-disk = %{version}-%{release}
+Requires: libvirt-daemon-driver-storage-logical = %{version}-%{release}
+Requires: libvirt-daemon-driver-storage-scsi = %{version}-%{release}
+Requires: libvirt-daemon-driver-storage-iscsi = %{version}-%{release}
+Requires: libvirt-daemon-driver-storage-mpath = %{version}-%{release}
+    %if %{with_storage_iscsi_direct}
+Requires: libvirt-daemon-driver-storage-iscsi-direct = %{version}-%{release}
+    %endif
+    %if %{with_storage_gluster}
+Requires: libvirt-daemon-driver-storage-gluster = %{version}-%{release}
+    %endif
+    %if %{with_storage_rbd}
+Requires: libvirt-daemon-driver-storage-rbd = %{version}-%{release}
+    %endif
+    %if %{with_storage_zfs}
+Requires: libvirt-daemon-driver-storage-zfs = %{version}-%{release}
+    %endif
 
-Requires:       %{name}-daemon-driver-storage-core = %{version}-%{release}
-Requires:       %{name}-libs = %{version}-%{release}
+%description daemon-driver-storage
+The storage driver plugin for the libvirtd daemon, providing
+an implementation of the storage APIs using LVM, iSCSI,
+parted and more.
 
-%description daemon-driver-storage-scsi
-The storage driver backend adding implementation of the storage APIs for scsi
-host devices.
+    %if %{with_qemu}
+%package daemon-driver-qemu
+Summary: QEMU driver plugin for the libvirtd daemon
+Requires: libvirt-daemon-common = %{version}-%{release}
+Requires: libvirt-daemon-log = %{version}-%{release}
+Requires: libvirt-libs = %{version}-%{release}
+%if 0%{?mariner}
+Requires: qemu-img
+%else
+Requires: /usr/bin/qemu-img
+%endif
+# For image compression
+Requires: gzip
+Requires: bzip2
+Requires: lzop
+Requires: xz
+%if 0%{!?mariner}
+Requires: systemd-container
+%endif
+%if 0%{!?mariner}
```

Comment:
dependency that isn't in mariner yet.  disabled for now.  not sure the impact.

Reply:
[swtpm fedora spec](https://[src.fedoraproject.org/rpms/swtpm/raw/main/f/swtpm.spec](https://src.fedoraproject.org/rpms/swtpm/raw/main/f/swtpm.spec))

Reply:
swtpm is merged into mariner, can uncomment this.